flac.go

```diff
@@ -403,20 +374,17 @@ func (stream *Stream) makeSeekTable() (err error) {
 	var sampleNum uint64
 	var tmp []meta.SeekPoint
 	for {
+		o, err := rs.Seek(0, io.SeekCurrent)
+		if err != nil {
+			return err
+		}
 		f, err := stream.ParseNext()
 		if err == io.EOF {
```

Comment:
Suggested rewrite, from:

```go
if err == io.EOF {
   break
}
if err != nil {
   return err
}
```

to:

```go
if err != nil {
   if err == io.EOF {
      break
   }
   return err
}
```

This makes the error branch contained to a single syntactic block, and thus gathers all different ways to handle the same error.